instrumentation/mysql/src/main/java/brave/mysql/TracingStatementInterceptor.java

```diff
@@ -50,7 +50,7 @@ public ResultSetInternalMethods preProcess(String sql, Statement interceptedStat
     if (interceptedStatement instanceof PreparedStatement) {
       sql = ((PreparedStatement) interceptedStatement).getPreparedSql();
     }
-    int spaceIndex = sql.indexOf(' '); // Allow span names of single-word statements like COMMIT
+    int spaceIndex = Math.min(sql.indexOf(' '), sql.indexOf('\n')); // Allow span names of single-word statements like COMMIT
```

Comment:
slick

Reply:
This is pretty inefficient since we need to iterate the string twice (and many queries won't have newline so it's a full scan of potentially very long ones).

Let's instead loop through the characters and return the first that matches

```
int findWhitespace(String str) {
for (int i = 0; i < str.length(); i++) {
  char c = str.charAt(i);
  switch (c) {
    case ' ':
    case '\n':
    case '\r':
    case '\t':
      return I;
    default:
      // Skip
 }
}
return -1;
}
```

Not sure how official it is but I found this grammar that lists those four as whitespace possibilities

https://github.com/antlr/grammars-v4/blob/master/mysql/Positive-Technologies/MySqlLexer.g4

Reply:
sgtm good point on long queries. I forgot literally these are full query expressions